nvtabular/loader/backend.py

```diff
@@ -22,9 +23,10 @@ def _augment_schema(
     cats=None,
     conts=None,
     labels=None,
-    sparse_names=None,
-    sparse_max=None,
-    sparse_as_dense=False,
+    padded_cols=None,
```

Comment:
We couldn't really figure out what this function was supposed to do without the renames, so we went for it. The function name starts with an `_` so we've appropriately signaled that external code shouldn't depend on its stability. The two places in the NVTabular that use it call it via argument order instead of specifying the names, so I understand the caution but I think we're okay.

Reply:
looks like it will be a safe change, and the names make things clearer :). We may consider removing this functon along with the other loader code to follow-up on [the promise here](https://github.com/NVIDIA-Merlin/NVTabular/blob/v1.8.1/nvtabular/loader/__init__.py#L23) in one of the upcoming releases

Reply:
And by that point we'll hopefully have updated the dataloader API to a point where this augment_schema function is no longer required either here or the copies [in Transformers4Rec](https://github.com/NVIDIA-Merlin/Transformers4Rec/blob/v0.1.16/merlin_standard_lib/utils/misc_utils.py#L243) and [in Merlin Models](https://github.com/NVIDIA-Merlin/models/blob/v0.11.0/merlin/models/loader/backend.py#L19). The existence of this function suggests to me that there's something missing from the dataloader API as it currently exists.

Reply:
I think the something that's missing is "transforms implemented as operators that provide schema tracking" 😺